### PR TITLE
Add reset all config for zwave node, Show default value per parameter

### PR DESCRIPTION
**Pull request by wendevlin**
## Proposed change

Add a reset all config parameter functionality at the bottom of the zwave node configration panel.
image

Add the default value to every parameter:
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR is related to issue or discussion:  [zwave-js/certification-backlog#39](https://github.com/zwave-js/certification-backlog/issues/39)
- This PR is related to issue or discussion:  [zwave-js/certification-backlog#40](https://github.com/zwave-js/certification-backlog/issues/40)
- This PR needs the backend PR merged: https://github.com/home-assistant/core/pull/125343

## Checklist

- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by wendevlin** [comment]
Still in draft, because core isn't ready yet.

File: src/translations/en.json [hunk 1]
```diff
@@ -4900,7 +4900,19 @@
             "set_param_queued": "The parameter change has been queued, and will be updated when the device wakes up.",
             "set_param_error": "An error occurred.",
             "parameter": "Parameter",
-            "bitmask": "Bitmask"
+            "bitmask": "Bitmask",
+            "reset_to_default": {
+              "button_label": "Reset to default configuration",
+              "dialog": {
+                "title": "Reset to default",
+                "text": "Are you sure you want to reset all configuration parameters of the device {deviceName} to their default values?",
```
Comment (piitaya) [inline src/translations/en.json:4908]:
```suggestion
                "text": "All device parameters will be reset to their default values",
```

Comment (piitaya) [reply src/translations/en.json:4908]: According to the other confirmation dialogs, the text should be a description of the action, not a question.

**Review by piitaya — CHANGES_REQUESTED** [review]
Nice 🙂 Feature-wise it's look ok for me!

I only have feedback about code style.
For dialog, we don't usually add `ha-dialog` in the render function.
For your use-case, you have two choices : 

1. use generic dialog (e.g. showConfirmationDialog)
```
This will open a confirmation dialog and return a promise with a boolean.

For example : 
```ts
 const confirm = await showConfirmationDialog(this, {
    title: this.hass.localize(
      "ui.panel.config.category.editor.confirm_delete"
    ),
    text: this.hass.localize(
      "ui.panel.config.category.editor.confirm_delete_text"
    ),
    confirmText: this.hass.localize("ui.common.delete"),
    destructive: true,
  });
  ```

2. build your own dialog 
For example : https://github.com/home-assistant/frontend/blob/dev/src/dialogs/restart/show-dialog-restart.ts

If you need more logic than just text and confirmation, you can do something similar as you did but in separate files so the dialog can be lazy loaded when needed.

For this feature, I would keep it simple and use a confirmation dialog with a snack message for loading, success or error.

**Review by AlCalzone — APPROVED** [review]
From the Z-Wave perspective the frontend looks good. Just needs to actually do something when pressing the reset button.

File: src/translations/en.json [hunk 2]
```diff
@@ -4902,7 +4902,20 @@
...
             "set_param_error": "An error occurred.",
             "parameter": "Parameter",
-            "bitmask": "Bitmask"
+            "bitmask": "Bitmask",
+            "default": "Default",
+            "reset_to_default": {
+              "button_label": "Reset to default configuration",
+              "dialog": {
+                "title": "Reset to default",
+                "text": "All device parameters will be reset to their default values",
```
Comment (piitaya) [inline src/translations/en.json:4911]:
```suggestion
                "text": "All device parameters will be reset to their default values.",
```

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts [hunk 3]
```diff
@@ -427,6 +465,57 @@ class ZWaveJSNodeConfig extends LitElement {
     ]);
   }
 
+  private async _openResetDialog(event: Event) {
+    const progressButton = event.currentTarget as HaProgressButton;
+
+    await showConfirmationDialog(this, {
+      title: this.hass.localize(
```
Comment (piitaya) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts:472]:
```suggestion
      destructive: true,
      title: this.hass.localize(
```
We can add `destructive` to have the main button colored in red and not focused by default.

**Review by piitaya — COMMENTED** [review]
Seems good to me ! I only put few suggestions to improve the dialog 🙂

File: src/translations/en.json (hunk 1 above)
Comment (wendevlin) [reply src/translations/en.json]: thanks, I updated it like this.

**Comment by wendevlin** [comment]
Split this PR to: 
- https://github.com/home-assistant/frontend/pull/21990\
- https://github.com/home-assistant/frontend/pull/21991

**Pull request closed without merging**